spec/support/fake_analytics.rb

```diff
@@ -112,12 +112,49 @@ def browser_attributes
   end
 
   failure_message do |actual|
-    <<~MESSAGE
-      Expected that FakeAnalytics would have received event #{event_name.inspect}
-      with #{attributes.inspect}.
+    matching_events = actual.events[event_name]
+    if matching_events&.length == 1 && attributes.instance_of?(Hash)
+      # We found one matching event. Let's show the user a diff of the actual and expected
+      # attributes
+      expected = attributes
+      actual = matching_events.first
+      message = ''
+      message += "expected: #{expected}\n"
+      message += "     got: #{actual}\n\n"
+      message += "Diff:#{differ.diff(actual, expected)}"
+      message
+    elsif matching_events&.length == 1 && attributes.instance_of?(RSpec::Matchers::BuiltIn::Include)
```

Comment:
Maybe `#present?` as a more elegant way of checking for non-nil, non-empty array?

```suggestion
    if matching_events.present? && attributes.instance_of?(Hash)
      # We found one matching event. Let's show the user a diff of the actual and expected
      # attributes
      expected = attributes
      actual = matching_events.first
      message = ''
      message += "expected: #{expected}\n"
      message += "     got: #{actual}\n\n"
      message += "Diff:#{differ.diff(actual, expected)}"
      message
    elsif matching_events.present? && attributes.instance_of?(RSpec::Matchers::BuiltIn::Include)
```

Reply:
The reason I check for `length == 1` is because if we log multiple instances of the same event to analytics (eg we call `analytics#idv_in_person_usps_proofing_results_job_enrollment_updated` five times) then `length` will be greater than 1. In that case I am not trying to do a diff across multiple hashes and it'll just fallback to the old view of displaying all of the events that analytics received